pkg/kubectl/cmd/config/use_context.go

```diff
@@ -82,10 +84,16 @@ func (o *useContextOptions) complete(cmd *cobra.Command) bool {
 	return true
 }
 
-func (o useContextOptions) validate() error {
+func (o useContextOptions) validate(config *clientcmdapi.Config) error {
 	if len(o.contextName) == 0 {
 		return errors.New("You must specify a current-context")
 	}
 
-	return nil
+	for _, context := range config.Contexts {
+		if context.Cluster == o.contextName {
```

Comment:
This should be `context.Name == o.contextName`. `context.Name` and `context.Cluster` happen to be the same in kubeconfig generated by `kube-up.sh` scripts (and gcloud as well) for simplicity/lethargy, but they aren't required to be.